hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HiveAvroSerializer.java

```diff
@@ -299,14 +296,26 @@ private Object serializePrimitive(PrimitiveObjectInspector fieldOI, Object struc
         String string = (String)fieldOI.getPrimitiveJavaObject(structFieldData);
         return new Utf8(string);
       case DATE:
-        return DateWritable.dateToDays(((DateObjectInspector)fieldOI).getPrimitiveJavaObject(structFieldData));
+        try {
+          Class<?> clazz = structFieldData.getClass();
+          return  clazz.getMethod("getDays").invoke(structFieldData);
```

Comment:
This has a significant impact on performance，
 Let's avoid using reflection every iteration。
Let's do the same thing as HiveUtils

Reply:
done